confidential-assets/src/chunkedAmount.ts

```diff
@@ -0,0 +1,109 @@
+import { AnyNumber } from "@aptos-labs/ts-sdk";
+
+/**
+ * Number of chunks for confidential balance
+ */
+const CHUNKS_COUNT = 8;
+
+/**
+ * Max bits of amount in a chunk for normalized confidential balance
+ */
+const CHUNK_BITS = 16;
+
+export class ChunkedAmount {
+  amount: bigint;
+
+  amountChunks: bigint[];
+
+  chunksCount = CHUNKS_COUNT;
+
+  chunkBits = CHUNK_BITS;
+
+  static CHUNKS_COUNT = CHUNKS_COUNT;
+
+  static CHUNK_BITS = CHUNK_BITS;
+
+  static CHUNK_BITS_BIG_INT = BigInt(ChunkedAmount.CHUNK_BITS);
+
+  static CHUNKS_COUNT_HALF = ChunkedAmount.CHUNKS_COUNT / 2;
```

Comment:
This feels awkward, what if the dev sets one of these values. Even we don't get it right right, like if they pass in `chunksCount` to the constructor we only update `chunksCount`, not  `CHUNKS_COUNT`.